src/aleph/model/pending.py

```diff
@@ -16,6 +16,7 @@ class PendingMessage(BaseClass):
         IndexModel([("source.chain_name", ASCENDING)]),
         #    IndexModel([("source.height", ASCENDING)]),
         IndexModel([("message.time", ASCENDING)]),
+        IndexModel([("retries", ASCENDING), ("message.time", ASCENDING)]),
```

Comment:
Wouldn't a partial index be relevant in this case? Most messages will have `retries = 0`. A partial index on `retries > 0` would make it much smaller and reduce the insertion time.